ADE7816.h

```diff
@@ -166,27 +166,32 @@ class ADE7816
         virtual int _readRegister(const unsigned int reg)=0;
 };
 
-class ADE7816_SPI : public virtual ADE7816
+class ADE7816_SPI
 {
 	public:
-        ADE7816_SPI(int pin=energy_CS);\
+        
+        ADE7816_SPI(uint8_t spi_bus);
```

Comment:
Please add default parameter to allow compatibility with esp8266 and other other Arduino implementations.